tests/Middleware/HttpCacheTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Middleware;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Nyholm\Psr7\Response;
+use Nyholm\Psr7\ServerRequest;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Http\Method;
+use Yiisoft\Yii\Web\Middleware\HttpCache;
+
+class HttpCacheTest extends TestCase
```

Comment:
Additional test ideas:

1. `If-Modified-Since` should be ignored if `If-None-Match` header present.
2. Whole set of tests for ETags.
3. Test that middleware only works for GET/HEAD requests. For other request methods it should do nothing.

Reply:
Yes, I will add these tests later...

Reply:
Tests done